power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/meta_data.hpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_META_DATA_HPP
+#define POWER_GRID_MODEL_CPP_META_DATA_HPP
+
+#include "power_grid_model_c/meta_data.h"
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+namespace power_grid_model_cpp {
+class MetaData {
+  public:
+    static Idx _n_datasets() {
+        Handle handle{};
+        auto const n_datasets = PGM_meta_n_datasets(handle.get());
+        handle.check_error();
+        return n_datasets;
+    }
+
+    static MetaDataset const* _get_dataset_by_idx(Idx idx) {
+        Handle handle{};
+        auto const dataset = PGM_meta_get_dataset_by_idx(handle.get(), idx);
+        handle.check_error();
+        return dataset;
+    }
+
+    static MetaDataset const* _get_dataset_by_name(std::string const& dataset) {
+        Handle handle{};
+        auto const dataset_aux = PGM_meta_get_dataset_by_name(handle.get(), dataset.c_str());
+        handle.check_error();
+        return dataset_aux;
+    }
+
+    static std::string const _dataset_name(MetaDataset const* dataset) {
+        Handle handle{};
+        auto const name = std::string(PGM_meta_dataset_name(handle.get(), dataset));
+        handle.check_error();
+        return name;
+    }
+
+    static Idx _n_components(MetaDataset const* dataset) {
+        Handle handle{};
+        auto const n_components = PGM_meta_n_components(handle.get(), dataset);
+        handle.check_error();
+        return n_components;
+    }
+
+    static MetaComponent const* _get_component_by_idx(MetaDataset const* dataset, Idx idx) {
+        Handle handle{};
+        auto const component = PGM_meta_get_component_by_idx(handle.get(), dataset, idx);
+        handle.check_error();
+        return component;
+    }
+
+    static MetaComponent const* _get_component_by_name(std::string const& dataset, std::string const& component) {
+        Handle handle{};
+        auto const component_aux = PGM_meta_get_component_by_name(handle.get(), dataset.c_str(), component.c_str());
+        handle.check_error();
+        return component_aux;
+    }
+
+    static std::string const _component_name(MetaComponent const* component) {
+        Handle handle{};
+        auto const component_aux = std::string(PGM_meta_component_name(handle.get(), component));
+        handle.check_error();
+        return component_aux;
+    }
+
+    static size_t _component_size(MetaComponent const* component) {
```

Comment:
Why is this a problem?

Reply:
it's not a real problem per se, but the community has moved towards using signed indices and therefore also signed sizes (`ssize`). However, because our C API exposes `size_t`, it's probably best to keep it as such, because changing a data type in the public interface is breaking.